### DISCUSSION TITLE
`rarray`-like virtual table implementation needs `Values::get_array()` -- but that's private

### DISCUSSION THREAD

**Discussion by kkovaacs** [rusqlite/rusqlite#1497] [Questions] [answered]
I'm trying to implement a vtable like `rarray` but with some missing features added (like indexing on rowid). Unfortunately `rusqlite::vtab::Values::get_array()` is private, so I can't just use it in my own code.

[This](https://github.com/rusqlite/rusqlite/blob/997f313a143db283a13deba3463ba8fd65e5c3a2/src/vtab/array.rs#L156) is the relevant part of the source where access to the array received as an argument requires calling `get_array` as far as I can see.

Is there a good reason why `get_array()` is private? If yes, do you have any suggestion on how I could work around this issue?

**Comment by gwenn** [comment, marked as answer]
For me,
you need (unsafe) access to [Values::args](https://github.com/rusqlite/rusqlite/blob/997f313a143db283a13deba3463ba8fd65e5c3a2/src/vtab/mod.rs#L716) or `args[idx]`
Because `Values::get_array` is specific to `rarray` implementation:
- `array::ARRAY_TYPE`
- `array::Array`

**Reply by kkovaacs** [reply]
Hmm, makes sense, thanks. I'll implement something along the lines of `get_array`.
